remoulade/api/main.py

```diff
@@ -0,0 +1,60 @@
+""" This file describe the API to get the state of messages """
+from flask import jsonify, make_response
+from flask.views import MethodView
+
+from remoulade.state import StateNamesEnum
+
+from .app import MessageAdmin
+
+admin = MessageAdmin()
+app = admin.get_app()
+
+
+class MessageAPI(MethodView):
+    """ /messages/all/
+        API to get all the messages storaged in the backend
+    """
+
+    def get(self):
+        data = jsonify(admin.all())
+        return make_response(data, 200)
+
+
+class MessageItemAPI(MethodView):
+    """
+      /messages/id/<message_id>
+      Return  the message  with a certain id
+    """
+
+    error = {"IdNotFound": {"errorCode": "idNotFound", "errorMessage": "Id not found"}}
+
+    def get(self, message_id):
+        data = jsonify(admin.get(message_id))
+        if data is None:
+            return make_response(jsonify(self.error["IdNotFound"]), 400)
+        return make_response(data, 200)
+
+
+class MessageStateAPI(MethodView):
+    """
+      /messages/state/<state>
```

Comment:
there should be only two routes  `/messages/states` and `/messages/state/{message_id}`

You should pass state as GET parameter. (/messages/states?state=abc)

```
from flask import request
state = request.args.get('state')
```